specification/sections/02-providers.md

```diff
@@ -111,59 +111,25 @@ ResolutionDetails<number> resolveNumberValue(string flagKey, number defaultValue
 ResolutionDetails<MyStruct> resolveStructureValue(string flagKey, MyStruct defaultValue, context: EvaluationContext, options: FlagEvaluationOptions);
 ```
 
-#### Context Transformation
+#### Provider hooks
 
-Feature flag management systems often define structures representing arbitrary contextual data pertaining to the runtime, user, or application. The context transformer defines a simple interface to transform the OpenFeature `evaluation context` to such a structure, mapping values appropriately.
-
-See [evaluation context](./03-evaluation-context.md).
+Feature flag management systems often need to transform the context structures the user provides or generally address lifecycle concerns. A `provider hook` exposes a mechanism for providers to register [`Hooks`](./04-hooks.md) to tap into various stages of the flag evaluation lifecycle.
 
 ##### Requirement 2.10
 
-> The provider interface **MAY** define a `context transformer` method or function, which can be optionally implemented in order to transform the `evaluation context` prior to flag value resolution.
-
-The OpenFeature `client` might apply the transformer function before passing the returned value (the `transformed context`) to the provider resolution methods, thus allowing the provider implementation to avoid implementing and calling such transformation logic repeatedly in flag value resolution methods.
-
-```typescript
-class MyProvider implements Provider {
-  //...
-
-  // implementation of context transformer
-  MyProviderContext transformContext(EvaluationContext context) {
-    return new MyProviderContext(context.email, context.ip, context.httpMethod);
-  }
+> The provider interface **MAY** define a `provider hook` mechanism which can be optionally implemented in order to add `hook` instances to the client.
```

Comment:
I think we need to state that the provider abstraction MUST support attaching hooks, though obviously the provider author doesn't need to attach them to their provider. I think having some implementations not supporting provider hooks will be problematic, just too inconsistent.